### PR TITLE
Fetch summary from traceLog

### PR DESCRIPTION
**Pull request by farhan-sauce**
Fetch time-durations of scripting, rendering, painting from traceLogs.

Screen Shot 2019-08-07 at 4 31 22 PM

example code
```
import Tracelib from '../src/index'
import JANK_TRACE_LOG from './__fixtures__/jankTraceLog.json'

const tasks = new Tracelib(JANK_TRACE_LOG)
const summary = tasks.getSummary()
console.log(summary)

# OUTPUT
{
  rendering: 847.373997092247,
  painting: 69.94999980926514,
  other: 9.896000564098358,
  scripting: 394.4800021648407,
  idle: 52.38300037384033,
  startTime: 289959855.634,
  endTime: 289961229.717
}
```

### CODE REVIEW

File: src/timelineModel/index.ts [hunk 1]
```diff
@@ -633,90 +633,58 @@ export default class TimelineModel {
...
-                cpuProfile.samples = pushAll(cpuProfile.samples, samples)
-                cpuProfile.timeDeltas = pushAll(
-                    cpuProfile.timeDeltas,
-                    eventData.timeDeltas || []
-                )
-                if (
-                    cpuProfile.samples.length !== cpuProfile.timeDeltas.length
-                ) {
+                const eventData = profileEvent.args['data']
+                if ('startTime' in eventData)
```
Comment (christian-bromann) [inline src/timelineModel/index.ts]: Can enforce the brackets? I intentionally added them so the code becomes more clear.

Comment (christian-bromann) [reply src/timelineModel/index.ts]: maybe worth adding an eslint rule to it

File: src/timelineModel/timelineModelFilter/timelineModelFilter.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import Event from '../../tracingModel/event'
+
+export default class TimelineModelFilter {
```
Comment (christian-bromann) [inline src/timelineModel/timelineModelFilter/timelineModelFilter.ts]: let's rename this file to `index.ts` to follow the naming structure

File: src/timelineModel/timelineModelFilter/timelineModelViewDelegate.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import TimelineSelection from './timelineSelection'
+import Event from '../../tracingModel/event'
+
+export default class TimelineModelViewDelegate {
```
Comment (christian-bromann) [inline src/timelineModel/timelineModelFilter/timelineModelViewDelegate.ts]: is this file needed?

**Review by christian-bromann — CHANGES_REQUESTED** [review]
some comments but super stoked that we get this data

**Review by christian-bromann — APPROVED** [review]

File: README.md [hunk 4]
```diff
@@ -13,6 +13,33 @@ To install the package to your project, run:
...
+import JANK_TRACE_LOG from './jankTraceLog.json'
+
+const tasks = new Tracelib(JANK_TRACE_LOG)
+const summary = tasks.getSummary()
+console.log(summary)
+
+/**
+ * output:
+ * {
+ *   rendering: 847.373997092247,
```
Comment (adambabik) [inline README.md:33]: It's in milliseconds?

File: src/common/settings.ts [hunk 5]
```diff
@@ -8,7 +8,9 @@ export default class Settings {
     public static moduleSetting(settingName: string): Settings {
         const setting = this._moduleSettings.get(settingName)
         if (!setting) {
-            throw new Error('No setting registered: ' + settingName)
+            return {
+                get: (): null => null
```
Comment (adambabik) [inline src/common/settings.ts:12]: What exactly is it?

File: src/index.ts [hunk 6]
```diff
@@ -16,4 +18,17 @@ export default class Tracelib {
         return this._timelineLoader.performanceModel.frames()
             .map(( { duration } ): number => calcFPS(duration))
     }
+
+    public getSummary(from?: number, to?: number): StatsObject {
+        this._timelineLoader.init()
```
Comment (adambabik) [inline src/index.ts]: I assume we made sure that `init` can be called multiple times on `TimelineLoader` and it's fine?

File: src/index.ts [hunk 7]
```diff
@@ -16,4 +18,17 @@ export default class Tracelib {
...
+        const performanceModel = this._timelineLoader.performanceModel
+        this._timelineDetailsView = new TimelineDetailsView(performanceModel.findMainTrack())
```
Comment (adambabik) [inline src/index.ts]: Do we need to assign this to `this`? Is it reused later?

File: src/timelineModel/performanceModel.ts [hunk 8]
```diff
@@ -78,6 +81,13 @@ export default class PerformanceModel {
                 this._tracingModel.minimumRecordTime() + entry.timeOffset / 1000 - this._recordStartTime
             )
         }
+        this._autoWindowTimes()
+    }
+
+    public findMainTrack(): Track {
+        return this._timelineModel
+            .tracks()
+            .find((track): any => track.type === TrackType.MainThread && track.forMainFrame && track.events.length)
```
Comment (adambabik) [inline src/timelineModel/performanceModel.ts]:
```suggestion
            .find((track): Track => track.type === TrackType.MainThread && track.forMainFrame && track.events.length)
```

File: src/timelineModel/performanceModel.ts [hunk 9]
```diff
@@ -87,15 +97,19 @@ export default class PerformanceModel {
...
         model.adjustTime(this._tracingModel.minimumRecordTime() + timeOffset / 1000 - this._recordStartTime)
     }
 
     /**
      * @return {!SDK.TracingModel}
      */
     public tracingModel(): TracingModel {
-        if (!this._tracingModel) throw 'call setTracingModel before accessing PerformanceModel'
+        if (!this._tracingModel) {
+            throw 'call setTracingModel before accessing PerformanceModel'
```
Comment (adambabik) [inline src/timelineModel/performanceModel.ts]:
```suggestion
            throw new Error('call setTracingModel before accessing PerformanceModel')
```

File: src/timelineModel/performanceModel.ts [hunk 10]
```diff
@@ -119,4 +133,66 @@ export default class PerformanceModel {
     public frameModel(): TimelineFrameModel {
         return this._frameModel
     }
+
+    public setWindow (option: {left: number, right: number}) : void {
```
Comment (adambabik) [inline src/timelineModel/performanceModel.ts]:
```suggestion
    public setWindow(option: {left: number, right: number}): void {
```

File: README.md (hunk 4 above)
Comment (christian-bromann) [reply README.md:33]: yes

File: src/common/settings.ts (hunk 5 above)
Comment (christian-bromann) [reply src/common/settings.ts:12]: this lib doesn't need this, I assume devtools is using it to handle some sort of application settings

File: src/index.ts (hunk 6 above)
Comment (christian-bromann) [reply src/index.ts]: I can move it up into the constructor

**Pull request merged**